proxy/envoy/config.go

```diff
@@ -802,3 +736,66 @@ func buildMgmtPortListeners(mesh *proxyconfig.MeshConfig, managementPorts model.
 
 	return listeners, clusters
 }
+
+// buildEgressHTTPRoutes builds virtual hosts for services found in egress rules.
+// In addition, routing rules that match with egress services are also applied.
+func buildEgressHTTPRoutes(mesh *proxyconfig.MeshConfig, instances []*model.ServiceInstance,
+	config model.IstioConfigStore, httpConfigs HTTPRouteConfigs) HTTPRouteConfigs {
+
+	// Convert all egress rules into services and add route rules if any
+	egressRules, errs := model.RejectConflictingEgressRules(config.EgressRules())
+
+	if errs != nil {
+		glog.Warningf("Rejected rules: %v", errs)
+	}
+
+	for _, rule := range egressRules {
+		for _, port := range rule.Ports {
+			protocol := model.Protocol(strings.ToUpper(port.Protocol))
+			intPort := int(port.Port)
+			servicePort := &model.Port{Name: "external-traffic-port", Port: intPort, Protocol: protocol}
+			destination := rule.Destination.Service
+			svc := &model.Service{Hostname: destination}
+			routes := buildDestinationHTTPRoutes(svc, servicePort, instances, config)
```

Comment:
How does this work with labelled endpoint routing? I assume the rule would have externhost in the destination service field. Is the weighted LB not supported?

Reply:
Sorry I missed this. No label based routing/weighted LB. What I wanted to support was DP, retries, timeouts and fault injection.